### ISSUE TITLE
Error while training TVSN: cannot run doafn:forward()

### ISSUE THREAD

**Issue by talkenig** [silverbottlep/tvsn#6]
Ran the script_train_doafn.sh successfully. Now trying to train TVSN by running script_train_tvsn.sh.
The only thing I changed is the category name to chair. I am getting an error when the code reaches the spot when it tried to run doafn:forward():

> tal_kenig_gmail_com@tal-ubuntu:~/data/tvsn/tvsn-master/tvsn/code$ ./script_train_tvsn.sh
> {
>   loss_layer : 3
>   beta1 : 0.5
>   nThreads : 8
>   saveFreq : 20
>   batchSize : 15
>   split : "train"
>   lossnet : "vgg16"
>   lambda1 : 100
>   modelString : "TVSN"
>   gpu : 0
>   data_dir : "../data/"
>   dataset : "tvsn"
>   pixel_weight : 0
>   debug : 0
>   modelDir : "../snapshots/"
>   iterG : 2
>   background : 0
>   resume : 0
>   iter_per_epoch : 1000
>   category : "chair"
>   maxEpoch : 100
>   lambda2 : 0.001
>   lr : 0.0001
>   imgscale : 256
>   tv_weight : 0.0001
> }
> Random Seed: 4784
> <gpu> using device 0
> loading lossnet for perceptual loss...
> Initializing data loader nthread:8
> Starting donkey with id: 2 seed: 4786
> table: 0x41212e40
> Starting donkey with id: 4 seed: 4788
> table: 0x415f8f60
> Starting donkey with id: 1 seed: 4785
> table: 0x41f4bb18
> Starting donkey with id: 5 seed: 4789
> table: 0x40dd9998
> Starting donkey with id: 7 seed: 4791
> table: 0x402568b8
> Starting donkey with id: 3 seed: 4787
> table: 0x41b4fe48
> Starting donkey with id: 6 seed: 4790
> table: 0x406dc918
> Starting donkey with id: 8 seed: 4792
> table: 0x4178ec30
> Loading train metadata
> Loading train metadata
> Loading train metadata
> Loading train metadata
> Loading train metadata
> Creating train metadata
> Creating train metadata
> Creating train metadata
> Loading train metadata
> Creating train metadata
> Loading train metadata
> Creating train metadata
> Creating train metadata
> Creating train metadata
> Loading train metadata
> Creating train metadata
> Dataset: tvsn nTotal: 698 nTrain: 558 nTest: 140
> loading pretrained doafn ...
> creating model...
> 83
> <gpu> using device 0
> /home/tal_kenig_gmail_com/torch/install/bin/luajit: ...enig_gmail_com/torch/install/share/lua/5.1/nn/Linear.lua:66: cublas runtime error : the GPU program failed to execute at /home/tal_kenig_gmail_com/torch/extra/cutorch/lib/THC/THCBlas.cu:246
> stack traceback:
>         [C]: in function 'addmm'
>         ...enig_gmail_com/torch/install/share/lua/5.1/nn/Linear.lua:66: in function 'func'
>         ...mail_com/torch/install/share/lua/5.1/nngraph/gmodule.lua:345: in function 'neteval'
>         ...mail_com/torch/install/share/lua/5.1/nngraph/gmodule.lua:380: in function 'forward'
>         train_tvsn.lua:355: in function 'get_batch'
>         train_tvsn.lua:375: in main chunk
>         [C]: in function 'dofile'
>         ..._com/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
>         [C]: at 0x56393b05ae50

My guess is that the input parameter {batch_im_in, batch_view_in} is incompatible with the pre-trained network for some reason. This can't be an environment thing, as the same net does make successful forward feeds during training, right?
I tried to use the pre-trained doafn network posted by silverbottlep, got the same problem. I suspected the batch size differences between the scripts may be the problem, so tried training TVSN with batch size of 25, so it failed again the same place but with a different error:

> tal_kenig_gmail_com@tal-ubuntu:~/data/tvsn/tvsn-master/tvsn/code$ ./script_train_tvsn.sh
> {
>   loss_layer : 3
>   beta1 : 0.5
>   nThreads : 8
>   saveFreq : 20
>   batchSize : 25
>   split : "train"
>   lossnet : "vgg16"
>   lambda1 : 100
>   modelString : "TVSN"
>   gpu : 0
>   data_dir : "../data/"
>   dataset : "tvsn"
>   pixel_weight : 0
>   debug : 0
>   modelDir : "../snapshots/"
>   iterG : 2
>   background : 0
>   resume : 0
>   iter_per_epoch : 1000
>   category : "chair"
>   maxEpoch : 100
>   lambda2 : 0.001
>   lr : 0.0001
>   imgscale : 256
>   tv_weight : 0.0001
> }
> Random Seed: 4839
> <gpu> using device 0
> loading lossnet for perceptual loss...
> Initializing data loader nthread:8
> Starting donkey with id: 1 seed: 4840
> table: 0x415b1a78
> Starting donkey with id: 4 seed: 4843
> table: 0x41007cb0
> Starting donkey with id: 7 seed: 4846
> table: 0x417215e8
> Starting donkey with id: 2 seed: 4841
> table: 0x41bddb70
> Starting donkey with id: 5 seed: 4844
> table: 0x4091d7e8
> Starting donkey with id: 8 seed: 4847
> table: 0x41b81ee0
> Starting donkey with id: 3 seed: 4842
> table: 0x419ee100
> Starting donkey with id: 6 seed: 4845
> table: 0x40c3a188
> Loading train metadata
> Loading train metadata
> Creating train metadata
> Creating train metadata
> Loading train metadata
> Loading train metadata
> Creating train metadata
> Creating train metadata
> Loading train metadata
> Loading train metadata
> Creating train metadata
> Creating train metadata
> Loading train metadata
> Creating train metadata
> Loading train metadata
> Creating train metadata
> Dataset: tvsn nTotal: 698 nTrain: 558 nTest: 140
> loading pretrained doafn ...
> creating model...
> 83
> <gpu> using device 0
> /home/tal_kenig_gmail_com/torch/install/bin/luajit: ...h/install/share/lua/5.1/cudnn/SpatialFullConvolution.lua:31: attempt to perform arithmetic on field 'groups' (a nil value)
> stack traceback:
>         ...h/install/share/lua/5.1/cudnn/SpatialFullConvolution.lua:31: in function 'resetWeightDescriptors'
>         ...h/install/share/lua/5.1/cudnn/SpatialFullConvolution.lua:105: in function 'func'
>         ...mail_com/torch/install/share/lua/5.1/nngraph/gmodule.lua:345: in function 'neteval'
>         ...mail_com/torch/install/share/lua/5.1/nngraph/gmodule.lua:380: in function 'forward'
>         train_tvsn.lua:355: in function 'get_batch'
>         train_tvsn.lua:375: in main chunk
>         [C]: in function 'dofile'
>         ..._com/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
>         [C]: at 0x557100536e50

Not sure how to continue and debug from here, also quite new to torch. Would appreciate if someone can help wit this issue.  @silverbottlep : Can you help with that?

**Comment by talkenig** [comment]
Update:
Modified the script to use net-epoch-180.t7 for the doafn network, and it is running without an error.
Now my guess is that there is some problem with the file save at the last epoch... Still interested in understanding the problem.
